distributed/tests/test_config.py

```diff
@@ -204,3 +206,60 @@ def test_logging_extended():
             """
 
         subprocess.check_call([sys.executable, "-c", code])
+
+
+def test_logging_mutual_exclusive():
+    """
+    Ensure that 'logging-file-config' and 'logging' have to be mutual exclusive.
+    """
+    config = {'logging': {'dask': 'warning'}, 'logging-file-config': '/path/to/config'}
+    with pytest.raises(RuntimeError):
+        initialize_logging(config)
+
+
+def test_logging_file_config():
+    """
+    Test `logging-file-config` logging configuration
+    """
+    logging_config_contents = """
+[handlers]
+keys=console
+
+[formatters]
+keys=simple
+
+[loggers]
+keys=root, foo, foo_bar
+
+[handler_console]
+class=StreamHandler
+level=INFO
+formatter=simple
+args=(sys.stdout,)
+
+[formatter_simple]
+format=%(levelname)s: %(name)s: %(message)s
+
+[logger_root]
+level=WARNING
+handlers=console
+
+[logger_foo]
+level=INFO
+handlers=console
+qualname=foo
+
+[logger_foo_bar]
+level=ERROR
+handlers=console
+qualname=foo.bar
+"""
+    with tempfile.NamedTemporaryFile() as logging_config:
+        logging_config.file.write(logging_config_contents)
+        logging_config.file.seek(0)
+        dask_config = {'logging-file-config': logging_config.name}
+        with new_config(dask_config):
```

Comment:
You should really use a separate process, as in the tests above. Otherwise, subsequent tests will get an unexpected logging configuration.